Sources/NIOSSH/SshAgent/Framing.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+/// SSH Agent protocol encodes integers in big endian
+/// This struct is only used to write the the length prefix for messages
+struct SshAgentBinaryIntegerEncodingStrategy: NIOBinaryIntegerEncodingStrategy {
```

Comment:
I believe we have a fixed-width integer encoding strategy already, which might be suitable for this use-case.